File: src/main/java/parser/EquipementsParser.java
package parser;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import domain.Equipment;

import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.Writer;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class EquipementsParser {

    private final String amulettesDir = getClass().getResource("/equipements/amulettes.json").getPath();
    private final String anneauxDir = getClass().getResource("/equipements/anneaux.json").getPath();
    private final String coiffesDir = getClass().getResource("/equipements/coiffes.json").getPath();
    private final String capesDir = getClass().getResource("/equipements/capes.json").getPath();
    private final String dofusDir = getClass().getResource("/equipements/dofus.json").getPath();
    private final String bottesDir = getClass().getResource("/equipements/bottes.json").getPath();
    private final String ceinturesDir = getClass().getResource("/equipements/ceintures.json").getPath();
    private final String tropheesDir = getClass().getResource("/equipements/trophees.json").getPath();
    private final String boucliersDir = getClass().getResource("/equipements/boucliers.json").getPath();

    public List<Equipment> deserializedEquipments(String dir) throws FileNotFoundException {
        Equipment[] el = new GsonBuilder().create().fromJson(new FileReader(dir), Equipment[].class);
        return Arrays.asList(el);
    }

    public void writeAmulettes() throws IOException {
        write(amulettesDir, Equipment.AMULETTE);
    }

    public void writeAnneaux() throws IOException {
        write(anneauxDir, Equipment.ANNEAU);
    }

    public void writeBottes() throws IOException {
        write(bottesDir, Equipment.BOTTES);
    }

    public void writeBoucliers() throws IOException {
        write(boucliersDir, Equipment.BOUCLIER);
    }

    public void writeCapes() throws IOException {
        write(capesDir, Equipment.CAPE);
    }

    public void writeCeintures() throws IOException {
        write(ceinturesDir, Equipment.CEINTURE);
    }

    public void writeCoiffes() throws IOException {
        write(coiffesDir, Equipment.CHAPEAU);
    }

    public void writeDofus() throws IOException {
        write(dofusDir, Equipment.DOFUS);
    }

    public void writeTrophees() throws IOException {
        write(tropheesDir, Equipment.TROPHEE);
    }

    public void writeInFile(List<Equipment> dofus, Writer dofusWriter, Gson gson) throws IOException {
        gson.toJson(dofus, dofusWriter);
        dofusWriter.flush();
        dofusWriter.close();
    }

    private void write(String objectDir, String type) throws IOException {
        List<Equipment> equipmentsJson = this.deserializedEquipments(getClass().getResource("/equipements.json").getPath());
        List<Equipment> equipments = new ArrayList<>();
        Writer write = new FileWriter(objectDir);
        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        equipmentsJson.forEach(e -> {
            if (e.getType().equals(type)) {
                e.parseName();
                equipments.add(e);
            }
        });

        writeInFile(equipments, write, gson);
    }
}

File: src/main/java/parser/EquipementsParser.kt
package parser

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import domain.Equipement
import domain.json.EquipementJson
import java.io.File

object EquipementsParser {

    private val amulettesPath = javaClass.getResource("/equipements/amulettes.json").path
    private val anneauxPath = javaClass.getResource("/equipements/anneaux.json").path
    private val coiffesPath = javaClass.getResource("/equipements/coiffes.json").path
    private val capesPath = javaClass.getResource("/equipements/capes.json").path
    private val dofusPath = javaClass.getResource("/equipements/dofus.json").path
    private val bottesPath = javaClass.getResource("/equipements/bottes.json").path
    private val ceinturesPath = javaClass.getResource("/equipements/ceintures.json").path
    private val tropheesPath = javaClass.getResource("/equipements/trophees.json").path
    private val boucliersPath = javaClass.getResource("/equipements/boucliers.json").path

    fun writeAmulettes(equipementsPath: String) {
        write(equipementsPath, amulettesPath, EquipementJson.AMULETTE)
    }

    fun writeAnneaux(equipementsPath: String) {
        write(equipementsPath, anneauxPath, EquipementJson.ANNEAU)
    }

    fun writeBottes(equipementsPath: String) {
        write(equipementsPath, bottesPath, EquipementJson.BOTTES)
    }

    fun writeBoucliers(equipementsPath: String) {
        write(equipementsPath, boucliersPath, EquipementJson.BOUCLIER)
    }

    fun writeCapes(equipementsPath: String) {
        write(equipementsPath, capesPath, EquipementJson.CAPE)
    }

    fun writeCeintures(equipementsPath: String) {
        write(equipementsPath, ceinturesPath, EquipementJson.CEINTURE)
    }

    fun writeCoiffes(equipementsPath: String) {
        write(equipementsPath, coiffesPath, EquipementJson.CHAPEAU)
    }

    fun writeDofus(equipementsPath: String) {
        write(equipementsPath, dofusPath, EquipementJson.DOFUS)
    }

    fun writeTrophees(equipementsPath: String) {
        write(equipementsPath, tropheesPath, EquipementJson.TROPHEE)
    }

    private fun write(equipementsPath: String, pathWhereToWrite: String, type: String) {
        val equipementsJson = jacksonObjectMapper().readValue<List<EquipementJson>>(File(equipementsPath))
        val equipements: MutableList<Equipement> = mutableListOf()
        equipementsJson.forEach {
            if (it.type == type) {
                equipements.add(it.toEquipement())
            }
        }
        println("EQUIPEMENT PARSER pour : $type , size : ${equipements.size}")
        jacksonObjectMapper().writeValue(File(pathWhereToWrite), equipements)
    }
}